packages/@aws-cdk/aws-batch/test/job-definition.test.ts

```diff
@@ -0,0 +1,118 @@
+            readOnly: true,
+            ulimits: new Array<Ulimit>(),
+            user: 'root',
+            vcpus: 2,
+            volumes: new Array<Volume>(),
+        },
+        nodeProps: {
+            count: 2,
+            mainNode: 1,
+            rangeProps: new Array<batch.INodeRangeProps>(),
+        },
+        parameters: {
+            foo: 'bar',
+        },
+        retryAttempts: 2,
+        timeout: Duration.seconds(30),
+    } as batch.JobDefinitionProps, overrideProps);
+};
+
+describe('When creating a Batch Job Definition', () => {
+    test('it should match all specified properties', () => {
+        // GIVEN
+        const stack = new cdk.Stack();
+
+        // WHEN
+        const jobDefProps = getJobProps(stack);
+        new batch.JobDefinition(stack, 'job-def', jobDefProps);
+
+        // THEN
+        expect(stack).toHaveResourceLike('AWS::Batch::JobDefinition', {
+            JobDefinitionName: jobDefProps.jobDefinitionName,
+            ContainerProperties: jobDefProps.containerProps ? {
+                Command: jobDefProps.containerProps.command,
+                Environment: [
+                    {
+                        Name: 'foo',
+                        Value: 'bar',
+                    },
+                ],
+                InstanceType: jobDefProps.containerProps.instanceType.toString(),
+                LinuxParameters: {},
+                Memory: jobDefProps.containerProps.memoryLimitMiB,
+                MountPoints: [],
+                Privileged: jobDefProps.containerProps.privileged,
+                ReadonlyRootFilesystem: jobDefProps.containerProps.readOnly,
+                Ulimits: [],
+                User: jobDefProps.containerProps.user,
+                Vcpus: jobDefProps.containerProps.vcpus,
+                Volumes: [],
+            } : undefined,
+            NodeProperties: jobDefProps.nodeProps ? {
+                MainNode: jobDefProps.nodeProps.mainNode,
+                NodeRangeProperties: [],
+                NumNodes: jobDefProps.nodeProps.count,
+            } : undefined,
+            Parameters: {
+                foo: 'bar',
+            },
+            RetryStrategy: {
+                Attempts: jobDefProps.retryAttempts,
+            },
+            Timeout: {
+                AttemptDurationSeconds: jobDefProps.timeout ? jobDefProps.timeout.toSeconds() : -1,
+            },
+            Type: 'container',
+        }, ResourcePart.Properties);
+    });
+
+    test('it should be possible to create one from an ARN', () => {
+        // GIVEN
+        const stack = new cdk.Stack();
+
+        // WHEN
+        const jobDefProps = getJobProps(stack);
+        const existingJob = new batch.JobDefinition(stack, 'job-def', jobDefProps);
+        const job = batch.JobDefinition.fromJobDefinitionArn(stack, 'job-def-clone', existingJob.jobDefinitionArn);
+
+        // THEN
+        expect(job.jobDefinitionArn).toEqual(existingJob.jobDefinitionArn);
+    });
+});
```

Comment:
General comment: I'm not a fan of helper methods in tests. I think it's better to use `before` to set things up:

```ts

describe('Batch Job Definition', () => {
    let stack: cdk.Stack;
    let jobDefProps: batch.JobDefinitionProps;

    beforeEach(() => {
        stack = new cdk.Stack();

        const jobRepo = new Repository(stack, 'job-repo');

        const role = new iam.Role(stack, 'job-role', {
            assumedBy: new iam.ServicePrincipal('batch.amazonaws.com'),
        });

        const linuxParams = new LinuxParameters(stack, 'job-linux-params', {
            initProcessEnabled: true,
            sharedMemorySize: 1,
        });

        jobDefProps = {
            jobDefinitionName: 'test-job',
            containerProps: {
                command: [ 'echo "Hello World"' ],
                environment: {
                    foo: 'bar',
                },
                jobRole: role,
                gpuCount: 1,
                image: EcrImage.fromEcrRepository(jobRepo),
                instanceType: InstanceType.of(InstanceClass.T2, InstanceSize.MICRO),
                linuxParams,
                memoryLimitMiB: 1,
                mountPoints: new Array<MountPoint>(),
                privileged: true,
                readOnly: true,
                ulimits: new Array<Ulimit>(),
                user: 'root',
                vcpus: 2,
                volumes: new Array<Volume>(),
            },
            nodeProps: {
                count: 2,
                mainNode: 1,
                rangeProps: new Array<batch.INodeRangeProps>(),
            },
            parameters: {
                foo: 'bar',
            },
            retryAttempts: 2,
            timeout: Duration.seconds(30),
        };
    });

    test('renders the correct CloudFormation resource', () => {
        new batch.JobDefinition(stack, 'JobDefinition', jobDefProps);

        expect(stack).toHaveResourceLike('AWS::Batch::JobDefinition', {
            JobDefinitionName: jobDefProps.jobDefinitionName,
            ContainerProperties: jobDefProps.containerProps ? {
                Command: jobDefProps.containerProps.command,
                Environment: [
                    {
                        Name: 'foo',
                        Value: 'bar',
                    },
                ],
                InstanceType: jobDefProps.containerProps.instanceType.toString(),
                LinuxParameters: {},
                Memory: jobDefProps.containerProps.memoryLimitMiB,
                MountPoints: [],
                Privileged: jobDefProps.containerProps.privileged,
                ReadonlyRootFilesystem: jobDefProps.containerProps.readOnly,
                Ulimits: [],
                User: jobDefProps.containerProps.user,
                Vcpus: jobDefProps.containerProps.vcpus,
                Volumes: [],
            } : undefined,
            NodeProperties: jobDefProps.nodeProps ? {
                MainNode: jobDefProps.nodeProps.mainNode,
                NodeRangeProperties: [],
                NumNodes: jobDefProps.nodeProps.count,
            } : undefined,
            Parameters: {
                foo: 'bar',
            },
            RetryStrategy: {
                Attempts: jobDefProps.retryAttempts,
            },
            Timeout: {
                AttemptDurationSeconds: jobDefProps.timeout ? jobDefProps.timeout.toSeconds() : -1,
            },
            Type: 'container',
        });
    });

    test('can be imported from an ARN', () => {
        const importedJob = batch.JobDefinition.fromJobDefinitionArn(stack, 'job-def-clone',
          'arn:aws:batch:us-east-1:123456789012:job-definition/job-def-name:1');

        // THEN
        expect(importedJob.jobDefinitionName).toEqual('job-def-name');
        expect(importedJob.jobDefinitionArn).toEqual('arn:aws:batch:us-east-1:123456789012:job-definition/job-def-name:1');
    });
});
```

Reply:
Got it 👌